mail_multicompany_force_company/models/ir_mail_server.py

```diff
@@ -0,0 +1,21 @@
+from odoo import api, models
+from odoo.osv import expression
+
+
+class IrMailServer(models.Model):
+    _inherit = "ir.mail_server"
+
+    @api.model
+    def search(self, args, offset=0, limit=None, order=None, count=False):
+        domain = []
+        if not self._context.get("show_all"):
```

Comment:
For such context keys it's safer to prefix them with something unique, so that it stands a lower chance of conflicting with something from another module or custom code. eg `mail_multicompany_[dont_]force_company=True`